src/commonMethods/commonFunctions.js

```diff
@@ -0,0 +1,397 @@
+  let localInterfaces = os.networkInterfaces()
+  let parsedIp = ip.split('.')
+  let interfaceList = Object.keys(localInterfaces)
+  for (var i = 0; i < interfaceList.length; i++) {
+    var interface1 = localInterfaces[interfaceList[i]]
+
+    for (var k = 0; k < interface1.length; k++) {
+      if (interface1[k].family === 'IPv6') continue //not supporting IPv6, this shouldn't be called in IPv6 mode as there is no need for private/lan ranges anymore
+      if (interface1[k].address !== undefined) {
+        var parsedLocalIp = interface1[k].address.split('.')
+
+        if (
+          parsedIp[0] === parsedLocalIp[0] &&
+          parsedIp[1] === parsedLocalIp[1] &&
+          parsedIp[2] === parsedLocalIp[2]
+        ) {
+          return interface1[k].address
+        }
+      }
+    }
+    if (i === interfaceList.length - 1)
+      throw new Error('Could not determine IP network of the server that belongs to: ' + ip)
+  }
+}
+
+/**
+ * This function is used to restart the framework
+ */
+export const restartFramework = function() {
+  return this.$sequence([
+    () => killProcess('WPEFramework'),
+    () => killProcess('WPEProcess'),
+    () => startFramework(),
+  ])
+}
+
+/**
+ * This function performs below operations on WebKitBrowser Plugin
+ *  - Deactivate
+ *  - Activate
+ *  - Resume
+ */
+export const webKitBrowserOps = function() {
+  return this.$sequence([
+    () => pluginDeactivate.call(this, constants.webKitBrowserPlugin),
+    () => pluginActivate.call(this, constants.webKitBrowserPlugin),
+    () => webKitBrowserActions.call(this, constants.resume),
+  ])
+}
+
+/**
+ * This function is used to get Plugin Info
+ * @param plugin
+ * @returns {Promise<AxiosResponse<any>>}
+ */
+export const getPluginInfo = function(plugin) {
+  return this.$http
+    .get(`http://${constants.host}:80/Service/${plugin}`)
+    .then(result => result)
+    .catch(err => err)
+}
+
+/**
+ * This function is used to get the controller UI
+ * @returns {Promise<AxiosResponse<any>>}
+ */
+export const getControllerUI = function() {
+  return this.$http
+    .get(`http://${constants.host}:80/Service/Controller/UI`)
+    .then(result => result)
+    .catch(err => err)
+}
+
+/**
+ * This function is used to get CpuLoad
+ * @returns {Promise<T>}
+ */
+export const getCpuLoad = function() {
+  this.$thunder.api.DeviceInfo.systeminfo()
+    .then(result => {
+      this.$data.write('cpuload', result.cpuload)
+    })
+    .catch(err => err)
+}
+
+/**
+ * This function is used to get Controller plugin data
+ * @returns {Promise<any> | Thenable<any> | PromiseLike<any>}
+ */
+export const getControllerPluginData = function() {
+  return this.$thunder.api.Controller.status().then(result => result)
+}
+
+/**
+ * This function is used to put request
+ * @param plugin
+ * @returns {Promise<AxiosResponse<any>>}
+ */
+export const putRequestForPlugin = function() {
+  return this.$http
+    .put(`http://${constants.host}:80/Service/Provisioning`)
```

Comment:
REST APIs are deprecated, should be done using JSONRPC. If API is not available in JSONRPC yet we should not be making test cases for it yet, instead track it somewhere as "not supported on JSONRPC" in a ticket.

Reply:
I could not find the location of JSONRPC for Provisioning Plugin. Please let me know the location of JSONRPC